src/transformers/pipelines/image_segmentation.py

```diff
@@ -105,23 +111,46 @@ def _forward(self, model_inputs):
-            model_outputs, model_outputs["target_size"], threshold=threshold, mask_threshold=0.5
-        )
-        raw_annotation = raw_annotations[0]
-
-        raw_annotation["masks"] *= 255  # [0,1] -> [0,255] black and white pixels
-
-        raw_annotation["scores"] = raw_annotation["scores"].tolist()
-        raw_annotation["labels"] = [self.model.config.id2label[label.item()] for label in raw_annotation["labels"]]
-        raw_annotation["masks"] = [self._get_mask_str(mask) for mask in raw_annotation["masks"].cpu().numpy()]
-
-        # {"scores": [...], ...} --> [{"score":x, ...}, ...]
-        keys = ["score", "label", "mask"]
-        annotation = [
-            dict(zip(keys, vals))
-            for vals in zip(raw_annotation["scores"], raw_annotation["labels"], raw_annotation["masks"])
-        ]
+        if hasattr(self.feature_extractor, "post_process_segmentation"):
+            # Panoptic
+            raw_annotations = self.feature_extractor.post_process_segmentation(
+                model_outputs, model_outputs["target_size"], threshold=threshold, mask_threshold=0.5
+            )
+            raw_annotation = raw_annotations[0]
+            raw_annotation["masks"] *= 255  # [0,1] -> [0,255] black and white pixels
+            raw_annotation["scores"] = raw_annotation["scores"].tolist()
+            raw_annotation["labels"] = [self.model.config.id2label[label.item()] for label in raw_annotation["labels"]]
+            raw_annotation["masks"] = [self._get_mask_str(mask) for mask in raw_annotation["masks"].cpu().numpy()]
+
+            # {"scores": [...], ...} --> [{"score":x, ...}, ...]
+            keys = ["score", "label", "mask"]
+            annotation = [
+                dict(zip(keys, vals))
+                for vals in zip(raw_annotation["scores"], raw_annotation["labels"], raw_annotation["masks"])
+            ]
+        else:
+            # Default logits
+            logits = model_outputs.logits
+            if len(logits.shape) != 4:
+                raise ValueError(f"Logits don't have expected dimensions, expected [1, N, H, W], got {logits.shape}")
+            # Softmax
+            logits = logits.log_softmax(dim=1)
+            b, n, h, w = logits.shape
+            num_labels = len(self.model.config.id2label)
+            if n != num_labels:
+                raise ValueError(
+                    f"Logits don't have expected dimensions, expected [1, {num_labels}, H, W], got {logits.shape}"
+                )
+            size = model_outputs["target_size"].tolist()[0]
+            logits_reshaped = torchvision.transforms.Resize(size)(logits)
+            classes = logits_reshaped.argmax(dim=1)[0]
+            annotation = []
+            for label_id in range(num_labels):
+                label = self.model.config.id2label[label_id]
+                mask = classes == label_id
+                score = ((mask * logits_reshaped[0, label_id]).sum() / mask.sum()).exp().item()
```

Comment:
This score is the best I could come up on the spot.

In my estimation this is an average probability overall all pixels affected to that class. (using arithmetic mean of logits).
Is that sound ?

Reply:
The score per pixel should be the softmax of the logits, I believe.

Reply:
Yes I use the log_softmax to get the logits.

Reply:
Sorry I was reading this wrong. I don't think we should return a score per label like this. It doesn't correspond to any metric I have seen on semantic segmentation, but I may have missed something. If you want to continue the analgoy with token classification being semantic segmentation, there should be one score per pixel, which is the probability corresponding to the predicted class.